core/web/apiv2/auth.py

```diff
@@ -82,30 +111,60 @@ async def __call__(self, user: User = Depends(get_current_user)) -> User:
 # API Endpoints
 router = APIRouter()
 
-
-@router.post("/token")
-async def login(response: Response, form_data: OAuth2PasswordRequestForm = Depends()):
-
-    if not YETI_AUTH:
-        user = UserSensitive.find(username="yeti")
-        if not user:
-            user = UserSensitive(username="yeti", admin=True)
-            user.set_password("yeti")
-            user.save()
-    else:
-        user = UserSensitive.find(username=form_data.username)
-        if not (user and user.verify_password(form_data.password)):
-            raise HTTPException(
-                status_code=status.HTTP_401_UNAUTHORIZED,
-                detail="Incorrect username or password",
-                headers={"WWW-Authenticate": "Bearer"},
-            )
-
-    access_token = create_access_token(
-        data={"sub": user.username}, expires_delta=ACCESS_TOKEN_EXPIRE_MINUTES
-    )
-    response.set_cookie(key="yeti_session", value=access_token, httponly=True)
-    return {"access_token": access_token, "token_type": "bearer"}
+# We only want certain endpoints to be defined depending on the auth module.
+if AUTH_MODULE == 'oidc':
+    @router.get('/oidc-login')
+    async def login_info(request: Request):
+        redirect_uri = request.url_for('oidc_callback')
+        redirect_uri = redirect_uri.replace(netloc='localhost:8000')
+        return await get_oauth_client().oidc.authorize_redirect(request, redirect_uri)
+
```

Comment:
Based on previous comment, not enclosed in try/except block. Is this expected? How do you resurface bad configuration leading to exception in the UI?

Reply:
Right now it would be an Error 500. I haven't dug into what kind of errors I could get, but given that this involves navigation to another site (the OIDC provider), catching and handling all possible exceptions / failure modes seemed a bit daunting. Given that there is really not much a random user can do here (OIDC has to be setup by a server admin), I think it's reasonable to fail hard. WDYT?

Reply:
Maybe we should raise a specific exception with a message like "Error while authenticating with OIDC provider. Please contact your server admin". Wdyt?